app/controllers/meals_controller.rb

```diff
@@ -3,48 +3,95 @@ class MealsController < ApplicationController
 
   def index
     @meals = policy_scope(Meal).where.not(sodium: nil)
-    if params[:query].present?
-      search = @meals.search_by_preferences(params[:query])
-      tag = @meals.tagged_with(params[:query])
-      join = search + tag
-      @results = join.uniq
-      @meals = @results.sort_by { |meal| meal.sodium }.first(3)
+    case current_user.goal
+    when current_user.goal = "healthy"
+      if params[:query].present?
+        search = @meals.search_by_preferences(params[:query])
+        tag = @meals.tagged_with(params[:query])
+        join = search + tag
+        @results = join.uniq
+        @meals = @results.sort_by { |meal| meal.sodium }.first(3)
+      else
+        @meals = @meals.sort_by { |meal| meal.sodium }.first(3)
+        @user = current_user
+      end
+    when current_user.goal = "lose_weight"
+      if params[:query].present?
+        search = @meals.search_by_preferences(params[:query])
+        tag = @meals.tagged_with(params[:query])
+        join = search + tag
+        @results = join.uniq
+        @meals = @results.sort_by { |meal| meal.calories }.first(3)
+      else
+        @meals = @meals.sort_by { |meal| meal.calories }.first(3)
+        @user = current_user
+      end
+    when current_user.goal = "gain_muscle"
+      if params[:query].present?
+        search = @meals.search_by_preferences(params[:query])
+        tag = @meals.tagged_with(params[:query])
+        join = search + tag
+        @results = join.uniq
+        @meals = @results.sort_by { |meal| -meal.proteins }.first(3)
+      else
+        @meals = @meals.sort_by { |meal| -meal.proteins }.first(3)
+        @user = current_user
+      end
+    when current_user.goal = "gain_weight"
+      if params[:query].present?
+        search = @meals.search_by_preferences(params[:query])
+        tag = @meals.tagged_with(params[:query])
+        join = search + tag
+        @results = join.uniq
+        @meals = @results.sort_by { |meal| -meal.fat }.first(3)
+      else
+        @meals = @meals.sort_by { |meal| -meal.fat }.first(3)
+        @user = current_user
+      end
     else
```

Comment:
i'll merge it tho ahah

Reply:
you're right, else is not necessary! I was testing if the else will work first before I gave all the conditions, then left it in after adding all the conditions--my bad!
If `current_user.goal = nil` I'm not sure what would happen, but I feel like it would go in the else statement (if it is still there). Let's test it tomorrow! (sorry, not super experimental at the moment)